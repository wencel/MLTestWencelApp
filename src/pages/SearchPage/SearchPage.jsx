import PropTypes from 'prop-types';
import SearchNavbar from 'components/SearchNavbar';

/**
 * Component to render the search page.
 *
 * @component
 * @example
 * return (
 *   <SearchPage redirectSearchItems={()=>{})} />
 * )
 */
const SearchPage = ({ redirectSearchItems }) => {
  return <SearchNavbar onSearch={redirectSearchItems} />;
};

SearchPage.propTypes = {
  /**
   * Action to redirect if the search box event is triggered
   */
  redirectSearchItems: PropTypes.func,
};

export default SearchPage;
