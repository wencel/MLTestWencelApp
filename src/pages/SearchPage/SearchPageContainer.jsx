import PropTypes from 'prop-types';
import React from 'react';
import { connect } from 'react-redux';
import { redirectSearchItems } from 'reducers/item/itemActions';
import SearchPage from './SearchPage';

const SearchPageContainer = ({ redirectSearchItems }) => {
  return <SearchPage redirectSearchItems={redirectSearchItems} />;
};

SearchPageContainer.propTypes = {
  redirctSearchItems: PropTypes.func,
};

const mapDispatchToProps = dispatch => ({
  redirectSearchItems: data => {
    dispatch(redirectSearchItems(data));
  },
});

export default connect(null, mapDispatchToProps)(SearchPageContainer);
