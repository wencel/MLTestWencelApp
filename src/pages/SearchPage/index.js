import SearchPage from './SearchPageContainer';
export default SearchPage;
