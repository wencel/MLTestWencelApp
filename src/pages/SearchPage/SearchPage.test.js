import React from 'react';
import { shallow } from 'enzyme';
import SearchPage from './SearchPage';

describe('SearchPage', () => {
  it('renders correctly', () => {
    const redirectSearchItems = jest.fn();
    const wrapper = shallow(
      <SearchPage redirectSearchItems={redirectSearchItems} />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
