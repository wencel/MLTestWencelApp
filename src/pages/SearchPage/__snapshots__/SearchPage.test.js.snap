// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchPage renders correctly 1`] = `ShallowWrapper {}`;
